app/src/main/java/com/elavon/converge/processor/ConvergeService.java

```diff
@@ -0,0 +1,123 @@
+package com.elavon.converge.processor;
+
+import android.util.Log;
+
+import com.elavon.converge.exception.ConvergeClientException;
+import com.elavon.converge.model.ElavonTransactionRequest;
+import com.elavon.converge.model.ElavonTransactionResponse;
+import com.elavon.converge.model.ElavonTransactionSearchRequest;
+import com.elavon.converge.model.ElavonTransactionSearchResponse;
+import com.elavon.converge.model.mapper.ConvergeMapper;
+
+import java.math.BigDecimal;
+import java.net.SocketTimeoutException;
+import java.util.Date;
+
+import javax.inject.Inject;
+
+public class ConvergeService {
+
+    protected final ConvergeClient convergeClient;
+    protected final int maxRetryCount;
+
+    @Inject
+    public ConvergeService(final ConvergeClient convergeClient, final int maxRetryCount) {
+        this.convergeClient = convergeClient;
+        this.maxRetryCount = maxRetryCount;
+    }
+
+    public void create(final ElavonTransactionRequest request, final ConvergeCallback<ElavonTransactionResponse> callback) {
+        create(request, callback, 0, new Date());
+    }
+
+    private void create(
+            final ElavonTransactionRequest request,
+            final ConvergeCallback<ElavonTransactionResponse> callback,
+            final int retryCount,
+            final Date originalTime) {
+
+        if (retryCount > maxRetryCount) {
```

Comment:
should this be >= ? since the starting count is 0

Reply:
0 when first attempt (not a retry) then from 1 to `maxRetryCount` for retries.
For example if maxRetryCount is set to 2, there will be 1 initial attempt + 2 retries.
If maxRetryCount is set to 0, there will be 1 initial attempt + no retries.